docs/tutorials/nextjs.mdx

```diff
@@ -0,0 +1,337 @@
+---
+id: nextjs
+title: NextJS
+sidebar_label: NextJS
+hide_title: true
+---
+
+&nbsp;
+
+# NextJS Application Tutorial
+
+:::tip What You'll Learn
+
+- How to set up and use Redux Toolkit with the NextJS framework
+
+:::
+
+:::info Prerequisites
+
+- Familiarity with [ES6 syntax and features](https://www.taniarascia.com/es6-syntax-and-feature-overview/)
+- Knowledge of React terminology: [JSX](https://reactjs.org/docs/introducing-jsx.html), [State](https://reactjs.org/docs/state-and-lifecycle.html), [Function Components, Props](https://reactjs.org/docs/components-and-props.html), and [Hooks](https://reactjs.org/docs/hooks-intro.html)
+- Understanding of [Redux terms and concepts](https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow)
+- Working through the [Quick strt guide](./quick-start.mdx) is recommended
+
+:::
+
+## Introduction
+
+NextJS is a server side rendering framework for React that presents some unique challenges for using Redux properly. These challenges include:
+
+- **Per-request safe Redux store creation** - A NextJS server can handle multiple requests simultaneously. This means that the Redux store should be created per request and that the store should not be shared across requests.
+- **SSR-friendly store hydration** - NextJS applications are rendered twice, first on the server and again on the client. Failure to render the same page contents on both the client and the server will result in a "hydration error". So the Redux store will have to be initialized on the server and then re-initialized on the client with the same data in order to avoid hydration issues.
+- **SPA routing support** - NextJS supports a hybrid model for client side routing. A customer's first page load will get an SSR result from the server. Subsequent page navigation will be handled by the client. This means that with a singleton store defined in the layout, route-specific data will need to be selectively reset on route navigation, while non-route-specific data will need to be retained in the store.
+- **Server caching friendly** - Recent versions of NextJS (specifically applications using the App Router architecture) support aggressive server caching. The ideal store architecture should be compatible with this caching.
+
+There are two architectures for a NextJS application; the Pages Router and the App Router. The Pages Router is the original architecture for NextJS.
+
+Using Redux with the Pages Router is well understood and handled properly by the (next-redux-wrapper)[https://github.com/kirill-konshin/next-redux-wrapper]. This tutorial will focus on the App Router architecture as it is the new default architecture option for NextJS.
+
+### How to Read This Tutorial
+
+This page assumes that you already have an exisiting NextJS application based on the App Router architecture.
+
```

Comment:
```suggestion

If you don't have one. run the following command to follow along 
```bash
npx create-next-app next-with-redux
```
```

Reply:
Given that this is more of a setup guide, maybe it's better to say "create a blank Next app, and also see the `next-with-redux` example as well".

Reply:
just found that `create-next-app` has [an official example for redux with the App Router with id `with-redux`](https://github.com/vercel/next.js/tree/canary/examples/with-redux)

It can be installed by `npx create-next-app --example with-redux`